diff --git a/src/b.ts b/src/b.ts
index 028b76393f627ed86cc0644180c559fff9468384..07d2c2e3c3a3806a9fdcc52dca97a85f3d79f825 100644
--- a/src/b.ts
+++ b/src/b.ts
@@ -1,5 +1,6 @@
 // file for microbenchmarking
 
+import { Buffer } from 'node:buffer'
 import { Writer } from './buffer-writer'
 import { serialize } from './index'
 import { BufferReader } from './buffer-reader'
diff --git a/src/buffer-reader.ts b/src/buffer-reader.ts
index 2305e130c930bbd33974d3f7232faeeebca765c1..12cc441e744fdabfd6ad85a1fe77ef7a43806d65 100644
--- a/src/buffer-reader.ts
+++ b/src/buffer-reader.ts
@@ -1,10 +1,12 @@
+import { Buffer } from 'node:buffer'
+
 const emptyBuffer = Buffer.allocUnsafe(0)
 
 export class BufferReader {
   private buffer: Buffer = emptyBuffer
 
   // TODO(bmc): support non-utf8 encoding?
-  private encoding: string = 'utf-8'
+  private encoding = 'utf-8' as const
 
   constructor(private offset: number = 0) {}
 
diff --git a/src/buffer-writer.ts b/src/buffer-writer.ts
index 756cdc9f3cd403b3a83c8eb04fe716b78b9cbc06..226b6b0845a1f525b73860963a4f00622bd3741d 100644
--- a/src/buffer-writer.ts
+++ b/src/buffer-writer.ts
@@ -1,4 +1,5 @@
 //binary data writer tuned for encoding binary specific to the postgres binary protocol
+import { Buffer } from 'node:buffer'
 
 export class Writer {
   private buffer: Buffer
diff --git a/src/inbound-parser.test.ts b/src/inbound-parser.test.ts
index 364bd8d95296e7e1cf91b804bc12c3973d7576dc..7c26570131f5f662ac10bf229ebf5a46e70965b1 100644
--- a/src/inbound-parser.test.ts
+++ b/src/inbound-parser.test.ts
@@ -1,3 +1,4 @@
+import { Buffer } from 'node:buffer'
 import buffers from './testing/test-buffers'
 import BufferList from './testing/buffer-list'
 import { parse } from '.'
diff --git a/src/index.ts b/src/index.ts
index 00491ff7f411f613c3db0db289d72c0c307860d8..35b80a5f0baadc012a96a28b79e67efbdb9ec60d 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -1,4 +1,5 @@
-import { BackendMessage, DatabaseError } from './messages'
+import { Buffer } from 'node:buffer'
+import { DatabaseError } from './messages'
 import { serialize } from './serializer'
 import { Parser, MessageCallback } from './parser'
 
diff --git a/src/messages.ts b/src/messages.ts
index c3fbbdd9bbde8ddd17b79d3e257b417be3682390..a3bc52d5b4f449a2e01bb69f34b62ea377265824 100644
--- a/src/messages.ts
+++ b/src/messages.ts
@@ -1,3 +1,5 @@
+import { Buffer } from 'node:buffer'
+
 export type Mode = 'text' | 'binary'
 
 export type MessageName =
diff --git a/src/outbound-serializer.test.ts b/src/outbound-serializer.test.ts
index f6669becd2fadf6cea4e8cb52446ab00cf30f0fe..7a1888fe3c7d4ac51f0f2950da2109a07683ab12 100644
--- a/src/outbound-serializer.test.ts
+++ b/src/outbound-serializer.test.ts
@@ -1,3 +1,4 @@
+import { Buffer } from 'node:buffer'
 import assert from 'assert'
 import { serialize } from './serializer'
 import BufferList from './testing/buffer-list'
diff --git a/src/parser.ts b/src/parser.ts
index 1ad4e3abdc7724e6abe6f47dd4fb52060fe1438c..53ff8f158b1dd9c6a97765a9084faaffe65c2da2 100644
--- a/src/parser.ts
+++ b/src/parser.ts
@@ -1,3 +1,5 @@
+import { Buffer } from 'node:buffer'
+
 import { TransformOptions } from 'stream'
 import {
   Mode,
@@ -27,7 +29,6 @@ import {
   NoticeMessage,
 } from './messages'
 import { BufferReader } from './buffer-reader'
-import assert from 'assert'
 
 // every message is prefixed with a single bye
 const CODE_LENGTH = 1
@@ -80,7 +81,7 @@ export class Parser {
   private bufferLength: number = 0
   private bufferOffset: number = 0
   private reader = new BufferReader()
-  private mode: Mode
+  readonly mode: Mode
 
   constructor(opts?: StreamOptions) {
     if (opts?.mode === 'binary') {
diff --git a/src/serializer.ts b/src/serializer.ts
index 07e2fe4985e506bd989bb741df6cafe29c82cf68..27af0e3f1c83d9e8255d2b985ec65bcfb4f63122 100644
--- a/src/serializer.ts
+++ b/src/serializer.ts
@@ -1,3 +1,4 @@
+import { Buffer } from 'node:buffer'
 import { Writer } from './buffer-writer'
 
 const enum code {
diff --git a/src/testing/buffer-list.ts b/src/testing/buffer-list.ts
index 15ac785cc3bf2285466b7405ab9361d305360eca..5069573b02edbce3b7e08bd39b192245ff1b253f 100644
--- a/src/testing/buffer-list.ts
+++ b/src/testing/buffer-list.ts
@@ -1,3 +1,5 @@
+import { Buffer } from 'node:buffer'
+
 export default class BufferList {
   constructor(public buffers: Buffer[] = []) {}
 
diff --git a/src/testing/test-buffers.ts b/src/testing/test-buffers.ts
index a4d49f32295c4e3fe16caba93c75bdb8e8dc245c..b848651728ff44730165da2c833d7e84609c45f2 100644
--- a/src/testing/test-buffers.ts
+++ b/src/testing/test-buffers.ts
@@ -1,4 +1,5 @@
 // https://www.postgresql.org/docs/current/protocol-message-formats.html
+import { Buffer } from 'node:buffer'
 import BufferList from './buffer-list'
 
 const buffers = {
