--- postgis-3.5.0/configure
+++ postgis-3.5.0-wasm/configure
@@ -18972,7 +18972,7 @@
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking PROJ version via compiled code" >&5
 $as_echo_n "checking PROJ version via compiled code... " >&6; }
-  if test "$cross_compiling" = yes; then :
+  if test "$cross_compiling_not_a_problem" = yes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
