--- postgresql-16.3/contrib/pgstattuple/pgstatindex.c	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/contrib/pgstattuple/pgstatindex.c	2024-07-02 08:00:53.593359177 +0200
@@ -656,9 +656,9 @@
 			stats.unused_pages++;
 		else if (PageGetSpecialSize(page) !=
 				 MAXALIGN(sizeof(HashPageOpaqueData)))
-			ereport(ERROR,
+			ereport(WARNING,
 					(errcode(ERRCODE_INDEX_CORRUPTED),
-					 errmsg("index \"%s\" contains corrupted page at block %u",
+					 errmsg("# 661(FATAL block=%d): index \"%s\" contains corrupted page at block %u", blkno,
 							RelationGetRelationName(rel),
 							BufferGetBlockNumber(buf))));
 		else
