--- postgresql-16.3/src/backend/utils/init/postinit.c	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/backend/utils/init/postinit.c	2024-06-25 06:37:06.712799061 +0200
@@ -714,6 +714,7 @@
  *		Be very careful with the order of calls in the InitPostgres function.
  * --------------------------------
  */
+
 void
 InitPostgres(const char *in_dbname, Oid dboid,
 			 const char *username, Oid useroid,
@@ -1239,6 +1240,32 @@
 		CommitTransactionCommand();
 }
 
+/* ========================================================================*/
+/*
+void
+ReInitPostgres(const char *in_dbname, Oid dboid,
+			 const char *username, Oid useroid,
+			 bool load_session_libraries,
+			 bool override_allow_connections,
+			 char *out_dbname)
+{
+    puts("ReInitPostgres:Begin");
+    InitPostgres(in_dbname, dboid, username, useroid, load_session_libraries, override_allow_connections, out_dbname);
+    puts("ReInitPostgres:End");
+}
+*/
+/* ========================================================================*/
+
+
+
+
+
+
+
+
+
+
+
 /*
  * Process any command-line switches and any additional GUC variable
  * settings passed in the startup packet.
@@ -1345,14 +1372,15 @@
 static void
 ShutdownPostgres(int code, Datum arg)
 {
+puts("# 1348: " __FILE__);
 	/* Make sure we've killed any active transaction */
 	AbortOutOfAnyTransaction();
-
 	/*
 	 * User locks are not released by transaction end, so be sure to release
 	 * them explicitly.
 	 */
 	LockReleaseAll(USER_LOCKMETHOD, true);
+puts("# 1356: " __FILE__);
 }
 
 
