--- postgresql-16.3/src/backend/catalog/index.c	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/backend/catalog/index.c	2024-06-19 09:25:59.097131118 +0200
@@ -2994,7 +2994,7 @@
 		indexInfo->ii_ParallelWorkers =
 			plan_create_index_workers(RelationGetRelid(heapRelation),
 									  RelationGetRelid(indexRelation));
-
+#if !defined(__EMSCRIPTEN__)
 	if (indexInfo->ii_ParallelWorkers == 0)
 		ereport(DEBUG1,
 				(errmsg_internal("building index \"%s\" on table \"%s\" serially",
@@ -3006,7 +3006,7 @@
 								 RelationGetRelationName(indexRelation),
 								 RelationGetRelationName(heapRelation),
 								 indexInfo->ii_ParallelWorkers)));
-
+#endif
 	/*
 	 * Switch to the table owner's userid, so that any index functions are run
 	 * as that user.  Also lock down security-restricted operations and
