--- postgresql/src/include/common/logging.h
+++ postgresql-wasm/src/include/common/logging.h
@@ -91,7 +91,6 @@
 void		pg_logging_increase_verbosity(void);
 void		pg_logging_set_pre_callback(void (*cb) (void));
 void		pg_logging_set_locus_callback(void (*cb) (const char **filename, uint64 *lineno));
-
 void		pg_log_generic(enum pg_log_level level, enum pg_log_part part,
 						   const char *pg_restrict fmt,...)
 			pg_attribute_printf(3, 4);
