--- postgresql-16.3/src/bin/pg_resetwal/pg_resetwal.c	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/bin/pg_resetwal/pg_resetwal.c	2024-06-27 08:14:26.593051940 +0200
@@ -330,7 +330,7 @@
 	 * -- any other user won't have sufficient permissions to modify files in
 	 * the data directory.
 	 */
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__EMSCRIPTEN__) && !defined(__wasi__)
 	if (geteuid() == 0)
 	{
 		pg_log_error("cannot be executed by \"root\"");
