--- postgresql-16.3/src/include/common/file_utils.h	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/include/common/file_utils.h	2024-06-19 09:25:58.738138847 +0200
@@ -27,10 +27,14 @@
 struct iovec;					/* avoid including port/pg_iovec.h here */
 
 #ifdef FRONTEND
+#if !defined(fsync_fname)
 extern int	fsync_fname(const char *fname, bool isdir);
+#endif
 extern void fsync_pgdata(const char *pg_data, int serverVersion);
 extern void fsync_dir_recurse(const char *dir);
+#if !defined(durable_rename)
 extern int	durable_rename(const char *oldfile, const char *newfile);
+#endif
 extern int	fsync_parent_path(const char *fname);
 #endif
 
