--- postgresql-16.3/src/backend/access/nbtree/nbtutils.c	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/backend/access/nbtree/nbtutils.c	2024-06-25 06:37:06.808797660 +0200
@@ -2732,7 +2732,7 @@
 			break;
 		}
 	}
-
+#if !defined(__EMSCRIPTEN__)
 	if (debugmessage)
 	{
 		if (allequalimage)
@@ -2742,6 +2742,6 @@
 			elog(DEBUG1, "index \"%s\" cannot use deduplication",
 				 RelationGetRelationName(rel));
 	}
-
+#endif
 	return allequalimage;
 }
