--- postgresql-16.2/src/include/libpq/be-fsstubs.h	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/include/libpq/be-fsstubs.h	2024-05-06 08:20:15.251406348 +0200
@@ -19,8 +19,10 @@
  * Probably these should have had the underscore-free names,
  * but too late now...
  */
-extern int	lo_read(int fd, char *buf, int len);
-extern int	lo_write(int fd, const char *buf, int len);
+#if !defined(__EMSCRIPTEN__) && !defined(__wasi__)
+extern int	lo_read3(int fd, char *buf, int len);
+extern int	lo_write3(int fd, const char *buf, int len);
+#endif
 
 /*
  * Cleanup LOs at xact commit/abort
