--- postgresql-16.2/src/backend/tcop/postgres.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/backend/tcop/postgres.c	2024-05-25 06:30:21.914641631 +0200
@@ -3988,7 +3988,11 @@
 #endif
 }
 
-
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+#define PG_MAIN
+#include PG_MAIN_INCLUDE
+#undef PG_MAIN
+#else
 /*
  * PostgresSingleUserMain
  *     Entry point for single user mode. argc/argv are the command line
@@ -4884,7 +4888,7 @@
 		}
 	}							/* end of input-reading loop */
 }
-
+#endif /* wasm */
 /*
  * Throw an error if we're a WAL sender process.
  *
