--- postgresql-16.2/src/interfaces/libpq/fe-auth.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/interfaces/libpq/fe-auth.c	2024-05-06 08:20:15.252406333 +0200
@@ -1170,6 +1170,7 @@
 pg_fe_getusername(uid_t user_id, PQExpBuffer errorMessage)
 {
 	char	   *result = NULL;
+#if !defined(__EMSCRIPTEN__)
 	const char *name = NULL;
 
 #ifdef WIN32
@@ -1202,7 +1203,9 @@
 	else if (errorMessage)
 		appendPQExpBuffer(errorMessage, "%s\n", pwdbuf);
 #endif
-
+#else
+	const char *name = WASM_USERNAME;
+#endif
 	if (name)
 	{
 		result = strdup(name);
