--- postgresql-16.3/src/interfaces/libpq/legacy-pqsignal.c	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/interfaces/libpq/legacy-pqsignal.c	2024-07-02 11:07:55.571482844 +0200
@@ -32,6 +32,7 @@
  * non-ENABLE_THREAD_SAFETY builds), so the incompatibility isn't
  * troublesome for internal references.
  */
+#if !defined(__EMSCRIPTEN__)
 pqsigfunc
 pqsignal(int signo, pqsigfunc func)
 {
@@ -55,3 +56,4 @@
 	return signal(signo, func);
 #endif
 }
+#endif // __EMSCRIPTEN__
