--- postgresql/src/backend/tcop/postgres.c
+++ postgresql-wasm/src/backend/tcop/postgres.c
@@ -4052,7 +4052,11 @@
 #endif
 }
 
-
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+#define PG_MAIN
+#include PG_MAIN_INCLUDE
+#undef PG_MAIN
+#else
 /*
  * PostgresSingleUserMain
  *     Entry point for single user mode. argc/argv are the command line
@@ -4948,7 +4952,7 @@
 		}
 	}							/* end of input-reading loop */
 }
-
+#endif /* wasm */
 /*
  * Throw an error if we're a WAL sender process.
  *
@@ -5067,7 +5071,7 @@
 					 (long) user.tv_usec,
 					 (long) sys.tv_sec,
 					 (long) sys.tv_usec);
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__wasi__)
 
 	/*
 	 * The following rusage fields are not defined by POSIX, but they're
