--- postgresql-16.3/src/include/storage/fd.h	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/include/storage/fd.h	2024-07-02 11:07:55.357485911 +0200
@@ -188,13 +188,17 @@
 extern int	pg_fdatasync(int fd);
 extern void pg_flush_data(int fd, off_t offset, off_t nbytes);
 extern int	pg_truncate(const char *path, off_t length);
-extern void fsync_fname(const char *fname, bool isdir);
+extern void fd_fsync_fname(const char *fname, bool isdir);
 extern int	fsync_fname_ext(const char *fname, bool isdir, bool ignore_perm, int elevel);
-extern int	durable_rename(const char *oldfile, const char *newfile, int elevel);
+extern int	fd_durable_rename(const char *oldfile, const char *newfile, int elevel);
 extern int	durable_unlink(const char *fname, int elevel);
 extern void SyncDataDirectory(void);
 extern int	data_sync_elevel(int elevel);
 
+#define durable_rename(oldfile, newfile, elevel) fd_durable_rename(oldfile, newfile, elevel)
+#define fsync_fname(fname, isdir) fd_fsync_fname(fname, isdir)
+
+
 /* Filename components */
 #define PG_TEMP_FILES_DIR "pgsql_tmp"
 #define PG_TEMP_FILE_PREFIX "pgsql_tmp"
