--- postgresql/src/backend/commands/event_trigger.c
+++ postgresql-wasm/src/backend/commands/event_trigger.c
@@ -642,6 +642,8 @@
 	List	   *runlist;
 	EventTriggerData trigdata;
 
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+#else
 	/*
 	 * Event Triggers are completely disabled in standalone mode.  There are
 	 * (at least) two reasons for this:
@@ -660,6 +662,7 @@
 	 */
 	if (!IsUnderPostmaster)
 		return;
+#endif
 
 	runlist = EventTriggerCommonSetup(parsetree,
 									  EVT_DDLCommandStart,
@@ -690,12 +693,15 @@
 	List	   *runlist;
 	EventTriggerData trigdata;
 
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+#else
 	/*
 	 * See EventTriggerDDLCommandStart for a discussion about why event
 	 * triggers are disabled in single user mode.
 	 */
 	if (!IsUnderPostmaster)
 		return;
+#endif
 
 	/*
 	 * Also do nothing if our state isn't set up, which it won't be if there
@@ -738,12 +744,15 @@
 	List	   *runlist;
 	EventTriggerData trigdata;
 
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+#else
 	/*
 	 * See EventTriggerDDLCommandStart for a discussion about why event
 	 * triggers are disabled in single user mode.
 	 */
 	if (!IsUnderPostmaster)
 		return;
+#endif
 
 	/*
 	 * Use current state to determine whether this event fires at all.  If
@@ -809,12 +818,15 @@
 	List	   *runlist;
 	EventTriggerData trigdata;
 
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+#else
 	/*
 	 * See EventTriggerDDLCommandStart for a discussion about why event
 	 * triggers are disabled in single user mode.
 	 */
 	if (!IsUnderPostmaster)
 		return;
+#endif
 
 	/*
 	 * Also do nothing if our state isn't set up, which it won't be if there
