--- postgresql/src/backend/tcop/postgres.c
+++ postgresql-wasm/src/backend/tcop/postgres.c
@@ -3988,7 +3988,11 @@
 #endif
 }
 
-
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+#define PG_MAIN
+#include PG_MAIN_INCLUDE
+#undef PG_MAIN
+#else
 /*
  * PostgresSingleUserMain
  *     Entry point for single user mode. argc/argv are the command line
@@ -4884,7 +4888,7 @@
 		}
 	}							/* end of input-reading loop */
 }
-
+#endif /* wasm */
 /*
  * Throw an error if we're a WAL sender process.
  *
