--- postgresql-16.3/src/interfaces/libpq/fe-exec.c	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/interfaces/libpq/fe-exec.c	2024-07-04 08:22:34.533326136 +0200
@@ -1694,8 +1694,15 @@
 	/* Don't try to send if we know there's no live connection. */
 	if (conn->status != CONNECTION_OK)
 	{
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+    puts("#1699 !!!");
+    conn->status = CONNECTION_OK;
+    conn->asyncStatus = PGASYNC_IDLE;
+    PQconnectPoll(conn);
+#else
 		libpq_append_conn_error(conn, "no connection to the server");
 		return false;
+#endif
 	}
 
 	/* Can't send while already busy, either, unless enqueuing for later */
