--- postgresql/src/backend/utils/init/miscinit.c
+++ postgresql-wasm/src/backend/utils/init/miscinit.c
@@ -143,7 +143,7 @@
 	 * children, but for consistency we make all postmaster child processes do
 	 * this.
 	 */
-#ifdef HAVE_SETSID
+#if defined(HAVE_SETSID) && !defined(__wasi__)
 	if (setsid() < 0)
 		elog(FATAL, "setsid() failed: %m");
 #endif
@@ -209,7 +209,7 @@
 	if (my_exec_path[0] == '\0')
 	{
 		if (find_my_exec(argv0, my_exec_path) < 0)
-			elog(FATAL, "%s: could not locate my own executable path",
+			elog(WARNING, "%s:212: could not locate my own executable path",
 				 argv0);
 	}
 
@@ -368,7 +368,7 @@
 	 *
 	 * XXX can we safely enable this check on Windows?
 	 */
-#if !defined(WIN32) && !defined(__CYGWIN__)
+#if !defined(WIN32) && !defined(__CYGWIN__) && !defined(__EMSCRIPTEN__) && !defined(__wasi__)
 	if (stat_buf.st_uid != geteuid())
 		ereport(FATAL,
 				(errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
@@ -388,7 +388,7 @@
 	 * be proper support for Unix-y file permissions.  Need to think of a
 	 * reasonable check to apply on Windows.
 	 */
-#if !defined(WIN32) && !defined(__CYGWIN__)
+#if !defined(WIN32) && !defined(__CYGWIN__) && !defined(__EMSCRIPTEN__) && !defined(__wasi__)
 	if (stat_buf.st_mode & PG_MODE_MASK_GROUP)
 		ereport(FATAL,
 				(errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
@@ -409,7 +409,7 @@
 	 * Suppress when on Windows, because there may not be proper support for
 	 * Unix-y file permissions.
 	 */
-#if !defined(WIN32) && !defined(__CYGWIN__)
+#if !defined(WIN32) && !defined(__CYGWIN__) && !defined(__EMSCRIPTEN__) && !defined(__wasi__)
 	SetDataDirectoryCreatePerm(stat_buf.st_mode);
 
 	umask(pg_mode_mask);
@@ -1223,7 +1223,13 @@
 		 * Think not to make the file protection weaker than 0600/0640.  See
 		 * comments below.
 		 */
+
+#if defined(__wasi__)
+printf("# 1228: CreateLockFile(%s) w+ (forced)\n", filename);
+		fd = fileno(fopen(filename, "w+"));
+#else
 		fd = open(filename, O_RDWR | O_CREAT | O_EXCL, pg_file_create_mode);
+#endif
 		if (fd >= 0)
 			break;				/* Success; exit the retry loop */
 
