--- postgresql-16.3/src/interfaces/libpq/legacy-pqsignal.c
+++ postgresql-16.3-wasm/src/interfaces/libpq/legacy-pqsignal.c
@@ -32,6 +32,7 @@
  * non-ENABLE_THREAD_SAFETY builds), so the incompatibility isn't
  * troublesome for internal references.
  */
+#if !defined(__EMSCRIPTEN__)
 pqsigfunc
 pqsignal(int signo, pqsigfunc func)
 {
@@ -55,3 +56,4 @@
 	return signal(signo, func);
 #endif
 }
+#endif // __EMSCRIPTEN__
