--- postgresql-16.3/src/include/libpq/be-fsstubs.h	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/include/libpq/be-fsstubs.h	2024-06-25 06:37:06.414803411 +0200
@@ -19,8 +19,10 @@
  * Probably these should have had the underscore-free names,
  * but too late now...
  */
-extern int	lo_read(int fd, char *buf, int len);
-extern int	lo_write(int fd, const char *buf, int len);
+#if !defined(__EMSCRIPTEN__) && !defined(__wasi__)
+extern int	lo_read3(int fd, char *buf, int len);
+extern int	lo_write3(int fd, const char *buf, int len);
+#endif
 
 /*
  * Cleanup LOs at xact commit/abort
