--- postgresql-16.2/src/include/storage/ipc.h	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/include/storage/ipc.h	2024-05-06 08:20:15.251406348 +0200
@@ -64,8 +64,11 @@
 /* ipc.c */
 extern PGDLLIMPORT bool proc_exit_inprogress;
 extern PGDLLIMPORT bool shmem_exit_inprogress;
-
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+extern void pg_proc_exit(int code);
+#else
 extern void proc_exit(int code) pg_attribute_noreturn();
+#endif
 extern void shmem_exit(int code);
 extern void on_proc_exit(pg_on_exit_callback function, Datum arg);
 extern void on_shmem_exit(pg_on_exit_callback function, Datum arg);
