--- postgresql/src/interfaces/libpq/fe-auth.c
+++ postgresql-wasm/src/interfaces/libpq/fe-auth.c
@@ -1170,6 +1170,7 @@
 pg_fe_getusername(uid_t user_id, PQExpBuffer errorMessage)
 {
 	char	   *result = NULL;
+#if !defined(__EMSCRIPTEN__)
 	const char *name = NULL;
 
 #ifdef WIN32
@@ -1202,7 +1203,9 @@
 	else if (errorMessage)
 		appendPQExpBuffer(errorMessage, "%s\n", pwdbuf);
 #endif
-
+#else
+	const char *name = getenv("PGUSER");
+#endif
 	if (name)
 	{
 		result = strdup(name);
