--- postgresql-16.2/contrib/pgstattuple/pgstatindex.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/contrib/pgstattuple/pgstatindex.c	2024-05-06 08:20:15.234406589 +0200
@@ -656,9 +656,9 @@
 			stats.unused_pages++;
 		else if (PageGetSpecialSize(page) !=
 				 MAXALIGN(sizeof(HashPageOpaqueData)))
-			ereport(ERROR,
+			ereport(WARNING,
 					(errcode(ERRCODE_INDEX_CORRUPTED),
-					 errmsg("index \"%s\" contains corrupted page at block %u",
+					 errmsg("# 661(FATAL block=%d): index \"%s\" contains corrupted page at block %u", blkno,
 							RelationGetRelationName(rel),
 							BufferGetBlockNumber(buf))));
 		else
