--- postgresql-16.2/src/backend/commands/dbcommands.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/backend/commands/dbcommands.c	2024-05-06 08:20:15.237406546 +0200
@@ -472,10 +472,12 @@
 	if (MakePGDirectory(dbpath) < 0)
 	{
 		/* Failure other than already exists or not in WAL replay? */
+        puts("# 475:" __FILE__);
 		if (errno != EEXIST || !isRedo)
 			ereport(ERROR,
 					(errcode_for_file_access(),
 					 errmsg("could not create directory \"%s\": %m", dbpath)));
+        puts("# 479:" __FILE__);
 	}
 
 	/*
