--- postgresql-16.2/src/interfaces/ecpg/preproc/ecpg.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/interfaces/ecpg/preproc/ecpg.c	2024-05-06 10:08:40.209755868 +0200
@@ -138,7 +138,7 @@
 
 	if (find_my_exec(argv[0], my_exec_path) < 0)
 	{
-		fprintf(stderr, _("%s: could not locate my own executable path\n"), argv[0]);
+		fprintf(stderr, _("%s:141: could not locate my own executable path\n"), argv[0]);
 		return ILLEGAL_OPTION;
 	}
 
