--- postgresql-16.2/src/backend/bootstrap/bootscanner.c	2024-02-05 22:52:20.000000000 +0100
+++ postgresql-16.2-wasm/src/backend/bootstrap/bootscanner.c	2024-05-06 08:20:52.984867770 +0200
@@ -1,4 +1,4 @@
-#line 2 "bootscanner.c"
+#line 1 "bootscanner.c"
 /*-------------------------------------------------------------------------
  *
  * bootscanner.l
@@ -23,7 +23,7 @@
 #include "bootparse.h"
 #include "utils/guc.h"
 
-#line 27 "bootscanner.c"
+#line 26 "bootscanner.c"
 
 #define  YY_INT_ALIGNED short int
 
@@ -826,7 +826,7 @@
 
 static int	yyline = 1;			/* line number for error reporting */
 
-#line 830 "bootscanner.c"
+#line 829 "bootscanner.c"
 #define YY_NO_INPUT 1
 /*
  * Keyword tokens return the keyword text (as a constant string) in boot_yylval.kw,
@@ -839,7 +839,7 @@
  *
  * String tokens return a palloc'd string in boot_yylval.str.
  */
-#line 843 "bootscanner.c"
+#line 842 "bootscanner.c"
 
 #define INITIAL 0
 
@@ -1060,7 +1060,7 @@
 #line 72 "bootscanner.l"
 
 
-#line 1064 "bootscanner.c"
+#line 1063 "bootscanner.c"
 
 	while ( /*CONSTCOND*/1 )		/* loops until end-of-file is reached */
 		{
@@ -1280,7 +1280,7 @@
 #line 125 "bootscanner.l"
 YY_FATAL_ERROR( "flex scanner jammed" );
 	YY_BREAK
-#line 1284 "bootscanner.c"
+#line 1283 "bootscanner.c"
 case YY_STATE_EOF(INITIAL):
 	yyterminate();
 
