--- postgresql-16.2/src/backend/storage/lmgr/proc.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/backend/storage/lmgr/proc.c	2024-05-06 08:20:15.241406489 +0200
@@ -307,7 +307,7 @@
 
 	if (MyProc != NULL)
 		elog(ERROR, "you already exist");
-
+puts("# 314: " __FILE__);
 	/* Decide which list should supply our PGPROC. */
 	if (IsAnyAutoVacuumProcess())
 		procgloballist = &ProcGlobal->autovacFreeProcs;
