--- postgresql-16.2/src/interfaces/ecpg/preproc/preproc.h	2024-02-05 22:52:24.000000000 +0100
+++ postgresql-16.2-wasm/src/interfaces/ecpg/preproc/preproc.h	2024-05-06 08:21:46.042099272 +0200
@@ -1,4 +1,4 @@
-/* A Bison parser, made by GNU Bison 3.7.5.  */
+/* A Bison parser, made by GNU Bison 3.8.2.  */
 
 /* Bison interface for Yacc-like parsers in C
 
@@ -16,7 +16,7 @@
    GNU General Public License for more details.
 
    You should have received a copy of the GNU General Public License
-   along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.  */
 
 /* As a special exception, you may create a larger work that contains
    part or all of the Bison parser skeleton and distribute that work
@@ -673,6 +673,8 @@
 
 extern YYSTYPE base_yylval;
 extern YYLTYPE base_yylloc;
+
 int base_yyparse (void);
 
+
 #endif /* !YY_BASE_YY_PREPROC_H_INCLUDED  */
