--- postgresql/configure
+++ postgresql-wasm/configure
@@ -12565,11 +12565,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char inflate ();
+char inflate (int a,int b);
 int
 main ()
 {
-return inflate ();
+return inflate (0,0);
   ;
   return 0;
 }
@@ -13096,11 +13096,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char xmlSaveToBuffer ();
+char xmlSaveToBuffer (int,int,int);
 int
 main ()
 {
-return xmlSaveToBuffer ();
+return xmlSaveToBuffer (0,0,0);
   ;
   return 0;
 }
@@ -13588,11 +13588,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char uuid_export ();
+char uuid_export (int,int,int,int);
 int
 main ()
 {
-return uuid_export ();
+return uuid_export (0,0,0,0);
   ;
   return 0;
 }
