--- postgresql/src/backend/storage/ipc/dsm_impl.c
+++ postgresql-wasm/src/backend/storage/ipc/dsm_impl.c
@@ -306,7 +306,7 @@
 
 		ereport(elevel,
 				(errcode_for_dynamic_shared_memory(),
-				 errmsg("could not resize shared memory segment \"%s\" to %zu bytes: %m",
+				 errmsg("# 309: could not resize shared memory segment \"%s\" to %zu bytes: %m",
 						name, request_size)));
 		return false;
 	}
@@ -350,7 +350,13 @@
 static int
 dsm_impl_posix_resize(int fd, off_t size)
 {
-	int			rc;
+	int			rc=0;
+
+fprintf(stderr, "!!!!!!!!!!! ret=%d dsm_impl_posix_resize(fd= %d, size= %jd)\n", rc, fd, size);
+if (rc)
+    abort();
+
+#if 0
 	int			save_errno;
 	sigset_t	save_sigmask;
 
@@ -404,8 +410,15 @@
 		sigprocmask(SIG_SETMASK, &save_sigmask, NULL);
 		errno = save_errno;
 	}
+#else
+    rc = (int)ftruncate(fd, size);
+#endif
 
-	return rc;
+#if defined(__wasi__)
+    printf("# 419: MAYBE CLANG BUG ftruncate(fd=%d, size=%jd) = %d\n", (int)fd, size, (int)rc);
+    return 0;
+#endif
+	return (int)rc;
 }
 
 #endif							/* USE_DSM_POSIX */
