--- postgresql/src/backend/storage/lmgr/proc.c
+++ postgresql-wasm/src/backend/storage/lmgr/proc.c
@@ -306,7 +306,11 @@
 		elog(PANIC, "proc header uninitialized");
 
 	if (MyProc != NULL)
-		elog(ERROR, "you already exist");
+#if defined(__wasi__)
+		elog(WARNING, "# 309: you already exist");
+#else
+		elog(ERROR, "# 309: you already exist");
+#endif
 
 	/* Decide which list should supply our PGPROC. */
 	if (IsAnyAutoVacuumProcess())
@@ -519,7 +523,7 @@
 		elog(PANIC, "proc header uninitialized");
 
 	if (MyProc != NULL)
-		elog(ERROR, "you already exist");
+		elog(ERROR, "# 522: you already exist");
 
 	/*
 	 * We use the ProcStructLock to protect assignment and releasing of
