--- postgresql/src/backend/commands/dbcommands.c
+++ postgresql-wasm/src/backend/commands/dbcommands.c
@@ -1776,8 +1776,9 @@
 	RequestCheckpoint(CHECKPOINT_IMMEDIATE | CHECKPOINT_FORCE | CHECKPOINT_WAIT);
 
 	/* Close all smgr fds in all backends. */
+#if !defined(__wasi__) && !defined(__EMSCRIPTEN__)
 	WaitForProcSignalBarrier(EmitProcSignalBarrier(PROCSIGNAL_BARRIER_SMGRRELEASE));
-
+#endif
 	/*
 	 * Remove all tablespace subdirs belonging to the database.
 	 */
