--- postgresql/src/interfaces/libpq/fe-misc.c
+++ postgresql-wasm/src/interfaces/libpq/fe-misc.c
@@ -567,9 +567,10 @@
 {
 	int			someread = 0;
 	int			nread;
-
+puts("------------------ pqReadData --------------- : " __FILE__);
 	if (conn->sock == PGINVALID_SOCKET)
 	{
+puts("# 573");
 		libpq_append_conn_error(conn, "connection not open");
 		return -1;
 	}
@@ -614,8 +615,16 @@
 
 	/* OK, try to read some data */
 retry3:
+#if defined(__wasi__)
+    puts(" # 619 : pqReadData->recvfrom_bc "  __FILE__);
+    nread = recvfrom_bc(conn->sock, conn->inBuffer + conn->inEnd, conn->inBufSize - conn->inEnd, 0, NULL, NULL);
+    printf("# 620: pqsecure_read(%d)-> rtt\n", nread);
+    if (!nread)
+        return 0;
+#else
 	nread = pqsecure_read(conn, conn->inBuffer + conn->inEnd,
 						  conn->inBufSize - conn->inEnd);
+#endif
 	if (nread < 0)
 	{
 		switch (SOCK_ERRNO)
@@ -638,6 +647,7 @@
 				goto definitelyFailed;
 
 			default:
+puts("# 642");
 				/* pqsecure_read set the error message for us */
 				return -1;
 		}
@@ -754,6 +764,7 @@
 
 	/* Come here if lower-level code already set a suitable errorMessage */
 definitelyFailed:
+puts("# 764: definitelyFailed");
 	/* Do *not* drop any already-read data; caller still wants it */
 	pqDropConnection(conn, false);
 	conn->status = CONNECTION_BAD;	/* No more connection to backend */
@@ -827,7 +838,12 @@
 		int			sent;
 
 #ifndef WIN32
+#if defined(__wasi__)
+        sent = send(conn->sock, ptr, len, 0);
+        printf("pqSendSome in progress %d/%d\n", sent, len);
+#else
 		sent = pqsecure_write(conn, ptr, len);
+#endif /* __wasi__ */
 #else
 
 		/*
@@ -861,6 +877,7 @@
 					/* Absorb input data if any, and detect socket closure */
 					if (conn->sock != PGINVALID_SOCKET)
 					{
+PDEBUG("# 868: pqReadData ???????????????????????????????????");
 						if (pqReadData(conn) < 0)
 							return -1;
 					}
@@ -960,7 +977,9 @@
 
 		return pqSendSome(conn, conn->outCount);
 	}
-
+#if defined(__wasi__)
+    sock_flush();
+#endif /* __wasi__ */
 	return 0;
 }
 
@@ -1038,6 +1057,9 @@
 static int
 pqSocketCheck(PGconn *conn, int forRead, int forWrite, time_t end_time)
 {
+#if defined(__wasi__)
+    return 1;
+#else
 	int			result;
 
 	if (!conn)
@@ -1071,6 +1093,7 @@
 	}
 
 	return result;
+#endif
 }
 
 
