--- postgresql/src/backend/access/transam/xlogarchive.c
+++ postgresql-wasm/src/backend/access/transam/xlogarchive.c
@@ -33,6 +33,10 @@
 #include "storage/ipc.h"
 #include "storage/lwlock.h"
 
+#if defined(__wasi__)
+#define system(cmd) system_wasi(cmd)
+#endif
+
 /*
  * Attempt to retrieve the specified file from off-line archival storage.
  * If successful, fill "path" with its complete path (note that this will be
