--- postgresql/src/backend/utils/fmgr/dfmgr.c
+++ postgresql-wasm/src/backend/utils/fmgr/dfmgr.c
@@ -121,12 +121,15 @@
 
 	/* Look up the function within the library. */
 	retval = dlsym(lib_handle, funcname);
-
+#if !defined(__wasi__)
 	if (retval == NULL && signalNotFound)
 		ereport(ERROR,
 				(errcode(ERRCODE_UNDEFINED_FUNCTION),
 				 errmsg("could not find function \"%s\" in file \"%s\"",
 						funcname, fullname)));
+#else
+    fprintf(stderr, "could not find function \"%s\" in file \"%s\" rv=%p snf=%b\n", funcname, fullname, retval, signalNotFound);
+#endif
 
 	pfree(fullname);
 	return retval;
