--- postgresql/src/include/port/emscripten.h
+++ postgresql-wasm/src/include/port/emscripten.h
@@ -0,0 +1,32 @@
+/* src/include/port/emscripten.h */
+
+#ifndef I_EMSCRIPTEN
+#define I_EMSCRIPTEN
+
+#if !defined(__cplusplus)
+#include <emscripten.h>
+#endif
+
+#include "/tmp/pglite/include/wasm_common.h"
+
+
+
+
+#define BOOT_END_MARK "build indices"
+#define FD_BUFFER_MAX 16384
+
+
+/*
+ * and now popen will return predefined slot from a file list
+ * as file handle in initdb.c
+ */
+
+#if defined(PG_LINKWEB)
+#if defined(PG_INITDB_MAIN) || defined(PG_POSTINIT)
+#define popen(cmd,type) pg_popen(cmd,type)
+extern FILE *pg_popen(const char *command, const char *type);
+#endif // PG_POSTINIT + PG_INITDB_MAIN
+#endif // PG_LINKWEB
+
+
+#endif // I_EMSCRIPTEN
