--- postgresql-16.3/src/backend/main/main.c	2024-05-06 22:21:25.000000000 +0200
+++ postgresql-16.3-wasm/src/backend/main/main.c	2024-06-27 08:14:26.478052884 +0200
@@ -41,20 +41,20 @@
 #include "utils/pg_locale.h"
 #include "utils/ps_status.h"
 
-
-const char *progname;
 static bool reached_main = false;
 
-
 static void startup_hacks(const char *progname);
 static void init_locale(const char *categoryname, int category, const char *locale);
 static void help(const char *progname);
 static void check_root(const char *progname);
 
-
 /*
  * Any Postgres server process begins execution here.
  */
+#if defined(__EMSCRIPTEN__) || defined(__wasi__)
+#include PG_MAIN_INCLUDE
+#else
+const char* progname;
 int
 main(int argc, char *argv[])
 {
@@ -199,6 +199,7 @@
 	/* the functions above should not return */
 	abort();
 }
+#endif /* wasm */
 
 
 
